File: Q16/Q16.java
import java.util.Scanner;

public class Q16 {
    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        int numberOfRows;

        // the loop is used to check if the number given can make a pattern
        while (true) {
            System.out.print("Enter a number: ");
            numberOfRows = scanner.nextInt();
            if (numberOfRows > 1)
                break;
            System.out.println("Invalid number. Please try again.");
        }

        // outer loop is for the number of rows
        // inner loop is for the number of asterisks per row
        for (int i = numberOfRows; i > 0; i--) {
            for (int j = 1; j <= i; j++)
                System.out.print("*");
            System.out.println();
        }

        // outer loop is for the number of rows
        // inner loop is for the number of asterisks per row
        for (int i = 2; i <= numberOfRows; i++) {
            for (int j = 1; j <= i; j++)
                System.out.print("*");
            System.out.println();
        }
    }
}

File: Q16/Q16.cpp
#include<iostream>
using namespace std;

void pattern(int n)
{
    for(int i=n;i>0;--i)
    {
        cout<<"\n";
        for(int j=1;j<=i;++j)
            cout<<"*";
    }       
    for(int i=2;i<=n;++i)
    {
        cout<<"\n";
        for(int j=1;j<=i;++j)
            cout<<"*";
    } 
}

int main()
{
    int n;
    cout<<"Enter no. of columns: ";
    cin>>n;
    pattern(n);
    return 0;
}

/*Eg: 
Enter no. of columns: 5
*****
****
***
**
*
**
***
****
*****

*/